src/main.c

```diff
@@ -1634,8 +1635,14 @@ int main(int argc, char *argv[]) {
     else
       imsg_error(" - double check the connections and try again\n");
 
-    if(str_eq(pgm->type, "serialupdi") || str_eq(pgm->type, "SERBB"))
+    if(str_eq(pgm->type, "serialupdi"))
       imsg_error(" - use -b to set lower baud rate, e.g. -b %d\n", baudrate? baudrate/2: 57600);
+    else if(str_eq(pgm->type, "buspirate_bb") || str_eq(pgm->type, "linuxgpio") ||
+      str_eq(pgm->type, "par") || str_eq(pgm->type, "SERBB")) {
+      imsg_error(" - use -i %s to set a longer delay (in microseconds) between each bit state change, e.g. -i 50\n",
+        bitclock? "instead of -B": "");
+      pgm->close(pgm);
```

Comment:
Looking more thoroughly into the main.c code, I now believe that running `pgm->close{pgm)` after a failed programming session wouldn't help here, as this is already done here
https://github.com/avrdudes/avrdude/blob/118625e18b3b902f5fa1cec9049d009526354740/src/main.c#L1882-L1891

Closing sth twice is not good practice, so I'd remove the `close()` line here.

This issue probably requires a more detailed analysis of the `-c linuxgpio` programmer code and/or implementing `-E` options for this programmer. I won't be able to do this, though, I hasten to add.